src/main/java/us/jonathans/use_case/get_leaderboard/GetLeaderboardInteractor.java

```diff
@@ -12,11 +14,11 @@ public GetLeaderboardInteractor(GetLeaderboardRepositoryInterface getLeaderboard
         this.getLeaderboardPresenter = getLeaderboardOutputBoundary;
     }
 
-    public Leaderboard getLeaderboard() {
+    public Leaderboard getLeaderboard() throws IOException, InterruptedException {
         return getLeaderboardRepository.getLeaderboard();
     }
 
-    public void execute(){
+    public void execute() throws IOException, InterruptedException {
```

Comment:
This is probably where you want to catch this exception and tell presenter to create an unsuccessful view (probably just send empty leaderboard data to the presenter)